### PR TITLE
Add remove function

### PR DESCRIPTION
**Pull request by Pi1998**
# Brach Changes 💯 

- Removed hardcoded items from the tasks array. 👍 
- Created a new JavaScript file for additional functionality. 🔢 
- Implemented a function to add new tasks. 🥇 
- Implemented a function to delete tasks. 🗡️ 
- Implemented a function to edit task descriptions. 🎓 
- New tasks now default to "completed: false" and are indexed based on array length. 😮‍💨 
- Deleting a task updates the remaining items' indexes to maintain order. ⬆️ 
- All changes to the To-Do List are now saved in local storage. 🏪 
- Updated README file 📚

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,27 +1,22 @@
...
+// Retrieve tasks from local storage or create an empty array if no tasks are stored
+let tasks = JSON.parse(localStorage.getItem(LOCAL_STORAGE_TASK_KEY)) || [];
+
+// Function to render the tasks list
 function renderTasks() {
-  const todoList = document.getElementById('tdl-lists');
-  todoList.innerHTML = '';
+  toDoList.innerHTML = '';
 
   tasks.forEach((task) => {
```
Comment (shahadat3669) [inline src/index.js:34]: 👍👍 Well done on implementing the add task function and adding a function to delete all completed tasks! However, there are a couple of additional requirements based on the project specifications:

- [ ] As per the project requirements, you should also implement a function for deleting a specific task (removing an element from the array). Each task should have an icon as in the design related to it, allowing users to delete individual tasks. Remember to store the updated tasks in the local storage to maintain data persistence.

- [ ] Similarly, you need to add a function for editing task descriptions, which is currently missing from your code. This function should enable users to edit the description of a specific task. After editing, ensure that you store the updated tasks in the local storage to preserve the changes.


https://github.com/Pi1998/to-do-list/assets/55840999/285bb053-29c6-4ae9-bdfd-81f15c3fe149


By incorporating these functions, your to-do list will be more comprehensive and fully aligned with the project requirements. Keep up the great work, and I'm excited to see your project progress with these additional features! 📝🔧🚀

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @Pi1998 , 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ No linter Errors
✅ Descriptive PR Title
✅ Descriptive PR Description


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DuaneDave — APPROVED** [review]
Hi @Pi1998 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Your design is pretty nice and pixel perfect
✔️ All expected functionality are implemented correctly

## Optional suggestions
- You have done exceptionally well in implementing all the features, but to improve accessibility, I recommend making the delete button visible by default instead of when an item is attempted to get edited

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by tugsem** [comment]
## Invalid Code Review Request ❌ 

Hi @Pi1998 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

<h2>You have submitted a project, that is already approved and belongs to the prior milestone. Please open a 
PR with the changes to the interactive list.</h2>

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by mckent05 — COMMENTED** [review]
Hi @Pi1998, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 
- Good job implementing all required functionality 👍 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**